metrics/graphite/graphite.go

```diff
@@ -150,16 +154,39 @@ func (e *emitter) dial() error {
 // Start will kick off a background goroutine to
 // call Flush once every interval.
 func (e *emitter) Start(interval time.Duration) error {
-	err := e.dial()
-	if err != nil {
-		return err
-	}
-	go func() {
-		for range time.Tick(interval) {
-			e.Flush()
+	var err error
+	e.start.Do(func() {
+		err = e.dial()
+		if err != nil {
+			return
 		}
-	}()
-	return nil
+		go func() {
+			t := time.Tick(interval)
```

Comment:
The underlying `time.Ticker` is leaked when this goroutine exits. Use `time.Newticker` and add `defer t.Stop` to ensure it isn't leaked.